lib/finch/http1/conn.ex

```diff
@@ -61,6 +62,16 @@ defmodule Finch.Conn do
   def open?(%{mint: nil}), do: false
   def open?(%{mint: mint}), do: HTTP1.open?(mint)
 
+  def idle_time(conn, unit \\ :native) do
+    idle_time = System.monotonic_time() - conn.last_checkin
+
+    System.convert_time_unit(idle_time, :native, unit)
+  end
+
+  def reusable?(conn) do
```

Comment:
I like this idea a lot, thanks for the contribution!

Maybe we should add a clause that would skip the check if max_idle_time is infinity?

Reply:
Currently, `:infinity` "just works" with `<` (I believe this is the reason it is idiomatic to use it for `timestamp`s and allowed by nimble options).

That said, if folks think it would be more intentional to explicitly skip the check when the value of `:infinity` is present, I'd be ok doing so.

Reply:
I'm sure that it works, but I was thinking of it more as a minor optimization, to skip the work done in `idle_time/2`

Reply:
Whoops, my email reply was out of the thread. Happy to make this update.

Reply:
Fixed in https://github.com/keathley/finch/pull/92/commits/c4644d1a4bc7be8a7535a31e38820af80659758c